### PR TITLE
[numpy] nonzero

### PR DESCRIPTION
**Pull request by tingying2020**
numpy operator nonzero

### CODE REVIEW

File: src/operator/numpy/np_nonzero_op.cu [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ * \file np_nonzero_op.cu
+*/
+
+#include "np_nonzero_op-inl.h"
+#include <cub/cub.cuh>
+
+namespace mxnet {
+namespace op {
+
+struct PrefixSumInit{
```
Comment (haojin2) [inline src/operator/numpy/np_nonzero_op.cu]: `struct PrefixSumInit {`

File: tests/python/gpu/test_operator_gpu.py [hunk 2]
```diff
@@ -2331,6 +2331,47 @@ def test_math():
...
+                mx_out = test_nonzero(x)
+                assert mx_out.shape == np_out.shape
+                assert_almost_equal(mx_out.asnumpy(), np_out, rtol, atol)
+
+                # Test imperative once again
+                mx_out = mx.npx.nonzero(x)
+                np_out = np.nonzero(x.asnumpy())
+                np_out = np.transpose(np_out)
+                assert_almost_equal(mx_out.asnumpy(), np_out, rtol, atol)
+
```
Comment (haojin2) [inline tests/python/gpu/test_operator_gpu.py]: One more blank line here.

File: tests/python/gpu/test_operator_gpu.py [hunk 3]
```diff
@@ -2331,6 +2331,47 @@ def test_math():
...
+                      (1,),
+                      (1, 1),
+                      (1, 2, 3),
+                      (1, 0),
+                      (2, 0, 3)
+                      ]:
+            for oneType in types:
+                if oneType is 'float16':
+                    rtol=1e-2
+                    atol=1e-2
```
Comment (haojin2) [inline tests/python/gpu/test_operator_gpu.py]: I don't think this operator need different tolerance levels for different input data types.

Comment (haojin2) [reply tests/python/gpu/test_operator_gpu.py]: And no need to duplicate the same test in `test_operator_gpu.py`

File: src/operator/numpy/np_nonzero_op.cu (hunk 1 above)
Comment (tingying2020) [reply src/operator/numpy/np_nonzero_op.cu]: Fixed, thank you.

File: tests/python/gpu/test_operator_gpu.py (hunk 3 above)
Comment (tingying2020) [reply tests/python/gpu/test_operator_gpu.py]: Done, thanks.

File: tests/python/gpu/test_operator_gpu.py (hunk 2 above)
Comment (tingying2020) [reply tests/python/gpu/test_operator_gpu.py]: Done.

**Pull request merged**